src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -93,7 +96,11 @@ public NeuralQueryBuilder(StreamInput in) throws IOException {
         super(in);
         this.fieldName = in.readString();
         this.queryText = in.readString();
-        this.modelId = in.readString();
+        if (isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport()) {
```

Comment:
please add a comment on what the new logic is. You may move it to separate private method `getModelId` and put it in the method's description

Reply:
Yes I will add the comment. Moving into method looks unecessory.